src/main/java/lotto/domain/WinningResult.java

```diff
@@ -1,23 +1,53 @@
 package lotto.domain;
 
+import lotto.context.Error;
+
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 public class WinningResult {
     private final LottoPaper winningPaper;
+    private LottoNum bonusNum;
 
     public WinningResult(String winningLottoNumbers) {
+        validWinningResult(winningLottoNumbers);
         winningPaper = new LottoPaper(Arrays.stream(winningLottoNumbers.split(","))
                 .map(String::trim)
                 .map(Integer::parseInt)
-                .map(integer -> new LottoNum(integer))
+                .map(LottoNum::of)
                 .collect(Collectors.toList()));
     }
 
-    public List<Integer> getWinninLottoNumberToIntegerList() {
+    private void validWinningResult(String winningLottoNumbers) {
+        if (winningLottoNumbers == null || "".equals(winningLottoNumbers)) {
+            throw new IllegalArgumentException(Error.ERROR_INPUT_NULL.getMsg());
+        }
+    }
+
+    public List<Integer> getWinningLottoNumberToIntegerList() {
         return winningPaper.getLottoNumbers().stream()
                 .map(LottoNum::getLottoNum)
                 .collect(Collectors.toList());
     }
+
+    public void setBonusNum(LottoNum lottoNum) {
+        this.bonusNum = lottoNum;
+    }
+
+    public WinningTable makeWinningTable(List<LottoPaper> lottoPapers) {
+        WinningTable winningTable = new WinningTable();
+
+        lottoPapers.forEach(lottoPaper -> {
+            winningTable.setAutoIncrementMatchCountResult(lottoPaper.getMatchCount(getWinningLottoNumberToIntegerList()),
+                    isBonus(lottoPaper));
+        });
+        return winningTable;
+    }
+
+    private boolean isBonus(LottoPaper lottoPaper) {
```

Comment:
보너스볼을 포함하고 있는지에 대한 네이밍이므로 is- 로 시작하는 것보다 has- 가 더 자연스럽다고 생각해요!
```suggestion
    private boolean hasBonusBall(LottoPaper lottoPaper) {
```